File: backend/app/schemas/claude.py
from typing import List, Optional

from pydantic import BaseModel, Field


class ClaudeSpawnRequest(BaseModel):
    args: List[str] = Field(default_factory=list)
    prompt: Optional[str] = None
    stdin: Optional[str] = None
    session_id: Optional[str] = None
    timeout_ms: Optional[int] = None


class ClaudeSessionResponse(BaseModel):
    session_id: str

File: backend-bun/src/schemas/claude.ts
import { z } from "zod";

export const ClaudeSpawnRequestSchema = z.object({
  args: z.array(z.string()).optional(),
  prompt: z.string().optional(),
  stdin: z.string().optional(),
  session_id: z.string().optional(),
  timeout_ms: z.number().optional(),
});

export type ClaudeSpawnRequestInput = z.infer<typeof ClaudeSpawnRequestSchema>;

export interface ClaudeSessionResponse {
  session_id: string;
}
